tools/lsp/language.rs

```diff
@@ -341,16 +341,42 @@ pub fn register_request_handlers(rh: &mut RequestHandler) {
         {
             let p = tk.parent();
             let gp = p.parent();
-            if p.kind() == SyntaxKind::QualifiedName
-                && gp.as_ref().map_or(false, |n| n.kind() == SyntaxKind::Element)
+
+            if p.kind() == SyntaxKind::DeclaredIdentifier
+                && gp.as_ref().map_or(false, |n| n.kind() == SyntaxKind::Component)
             {
-                let range = util::node_to_lsp_range(&p);
+                let element = gp.as_ref().unwrap().child_node(SyntaxKind::Element).unwrap();
+
                 ctx.server_notifier.send_message_to_preview(
                     common::LspToPreviewMessage::HighlightFromEditor {
                         url: Some(uri),
-                        offset: gp.unwrap().text_range().start().into(),
+                        offset: element.text_range().start().into(),
                     },
                 );
+
+                let range = util::node_to_lsp_range(&p);
+                return Ok(Some(vec![lsp_types::DocumentHighlight { range, kind: None }]));
+            }
+
+            if p.kind() == SyntaxKind::QualifiedName
+                && gp.as_ref().map_or(false, |n| n.kind() == SyntaxKind::Element)
+            {
+                let range = util::node_to_lsp_range(&p);
+
+                if gp
+                    .as_ref()
+                    .unwrap()
+                    .parent()
+                    .as_ref()
+                    .map_or(false, |n| n.kind() != SyntaxKind::Component)
```

Comment:
why do you filter that?
Technically, the part being inherited should also be highlighted.

Reply:
In 

```slint
export component Foo inherits Bar { }
```

this (document) highlights both `Foo` and `Bar` independently or each other now, just as it did before.

It (live-preview-selection) highlights on `Foo` now, not on `Bar` now, as requested in #6409. I think that makes a lot of sense: The `Bar` is not really there, it is just the basis `Foo` is built on.